tempesta_fw/sock.c

```diff
@@ -1036,45 +1039,283 @@ ss_connect(struct sock *sk, struct sockaddr *uaddr, int uaddr_len, int flags)
 		goto out;
 	err = 0;
 out:
-	release_sock(sk);
+	bh_unlock_sock(sk);
 	return err;
 }
 EXPORT_SYMBOL(ss_connect);
 
 /*
- * The original functions are inet_bind() and inet6_bind().
- * These two can be made a bit shorter should that become necessary.
+ * Bind for IPv4 protocol. The original functions are inet_bind().
+ *
+ * NOTE: This code assumes that both IPv4 and IPv6 are compiled in as
+ * part of the Linux kernel, and not as separate loadable kernel modules.
```

Comment:
I don't think this comment is relevant here (as well as below). It's relevant for `ss_inet_create()` as the original implementation tries to load the module if the code is not in the kernel yet, while the Sync Sockets implementation skips that.